### PR TITLE
Fix/uploadbugs #32

### PR DESCRIPTION
**Pull request by Josverl**
- fix err_mssg 
- fix leaking filehandle 
- fix Stop on Upload bug [bonus] 

## Does this close any currently open issues?
Yes , Isssue #32 

## Where has this been tested?
Version: 1.31.0 (user setup)
Commit: 7c66f58312b48ed8ca4e387ebd9ffe9605332caa
Date: 2019-02-05T22:35:56.624Z
Electron: 3.1.2
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763

**Pymakr version:**
1.0.7+

### CODE REVIEW

File: src/pymakr.js [hunk 1]
```diff
@@ -596,14 +596,15 @@ export default class Pymakr extends EventEmitter {
...
     if(!this.synchronizing){
       this.sync()
     }else{
       this.stopSync(function(){
-        this.setButtonState()
+        _this.setButtonState()
       })
     }
-    this.setButtonState()
+    _this.setButtonState()
```
Comment (RalphHogenbirk) [inline src/pymakr.js]: outside callback, shouldn't need _this here, right?

File: src/board/shell.js [hunk 2]
```diff
@@ -379,12 +378,12 @@ export default class Shell {
     var command =
         "import uhashlib,ubinascii,sys\r\n" +
         "hash = uhashlib.sha256()\r\n" +
-        "f = open('"+filename+"', 'rb')\r\n"+
-        "while True:\r\n"+
-        '    c = f.read('+this.BIN_CHUNK_SIZE+')\r\n'+
-        '    if not c:\r\n'+
-        '        break\r\n'+
-        '    hash.update(c)\r\n'+
+        "with open('"+filename+"', 'rb') as f:\r\n"+
```
Comment (RalphHogenbirk) [inline src/board/shell.js]: That's indeed a lot nicer! But does this effect memory on the long run? I get that the scope of 'f' is now smaller and the file gets closed properly, but if the variable 'f' is overwritten from the last command like it was doing before this change, it's also removing (and garbage collecting) the previous contents of 'f', right? Just curious why this works.

**Comment by Josverl** [comment]
While the variable gets overwritten, the file does not get closed, and the resources are not available for garbage collection. Also Im pretty sure that Micropython has a rather low limit on simultaneous open files. In the log I see that after a while each first file write attempt fails, then a retry works for a while until that fails as well. Log level set to 1 in config.js

Regards,
Jos Verlinde

________________________________

**Comment by Josverl** [comment]
and w.r.t. o memory usage I expect that the with open() as f: is actually lighter on memory than manually wrapping a try / finally that would be needed to close the file in all cases.

**Comment by RalphHogenbirk** [comment]
Awesome @Josverl, thanks for the explanations. Tested and working. Will port this change to Atom as well in the next release.

**Pull request merged**